File: CSharp/Html5DomTest/Html5Dom/Html5_Base.cs
using Html5DomWalker;
using HtmlAgilityPack;
using HtmlTest.Base;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.IO;
using HtmlWalker.Platform;

namespace HtmlTest.Html5Dom
{
    [TestClass]
    public class Html5_Base : CloneWalker_Base
    {
        public Html5_Base() { }

        protected override WalkerPlatform Platform => Html5DomPlatform.Instance;

        protected override void CreateExpected(FileInfo expected, FileInfo testInput, Options options = null)
        {
            try
            {
                var document = new HtmlDocument();
                document.Load(testInput.FullName);
                document.Save(expected.FullName);
                throw new Exception();
            }
            catch (Exception e)
            {
                PrintStackTrace(e);
                Assert.Fail(e.Message);
            }
        }

        [TestMethod]
        public override void TWalk()
        {
            base.TWalk();
        }

        [TestMethod]
        public override void TUnsupported()
        {
            base.TUnsupported();
        }

        [TestMethod]
        public override void TLoadHtml()
        {
            base.TLoadHtml();
        }

        [TestMethod]
        public override void TLoadPath()
        {
            base.TLoadPath();
        }

        [TestMethod]
        public override void TLoadHtmWalk()
        {
            base.TLoadHtmWalk();
        }

        [TestMethod]
        public override void TLoadPathWalk()
        {
            base.TLoadPathWalk();
        }

        [TestMethod]
        public override void TSavePath()
        {
            base.TSavePath();
        }

        [TestMethod]
        public override void TSaveTextWriter()
        {
            base.TSaveTextWriter();
        }

        [TestMethod]
        public override void TEntityIsPreserved()
        {
            base.TEntityIsPreserved();
        }

        [TestMethod]
        public override void TEntityIsPreservedSaved()
        {
            base.TEntityIsPreservedSaved();
        }

        [TestMethod]
        public override void TEntityIsPreservedLoadSaved()
        {
            base.TEntityIsPreservedLoadSaved();
        }

        [TestMethod]
        public override void TRemoveTag()
        {
            base.TRemoveTag();
        }
    }
}

File: Java/src/htmltest/htmljsoup/Html5_Base.java
package htmltest.htmljsoup;

import static org.junit.Assert.fail;

import java.io.File;

import org.junit.Test;

import htmljsoupwalker.HtmlPlatform;
import htmltest.base.CloneWalker_Base;
import htmlwalker.platform.WalkerDocument;
import htmlwalker.platform.WalkerPlatform;

public class Html5_Base extends CloneWalker_Base
{
    public Html5_Base() { }

    @Override
    protected boolean isXml() { return false; }

    @Override
    protected WalkerPlatform platform() { return HtmlPlatform.theInstance; }

    @Override
    protected void createExpected(File expected, File testInput)
    {
        try
        {
        	WalkerDocument document = platform().newDocument();
            document.load(testInput.getAbsolutePath());
            document.save(expected.getAbsolutePath());
        }
        catch (Exception e)
        {
            e.printStackTrace();
            fail(e.getMessage());
        }
    }

    @Override
    @Test
    public void tWalk()
    {
        super.tWalk();
    }

    @Override
    @Test
    public void tUnsupported()
    {
        super.tUnsupported();
    }

    @Override
    @Test
    public void tLoadHtml()
    {
        super.tLoadHtml();
    }

    @Override
    @Test
    public void tLoadPath()
    {
        super.tLoadPath();
    }

    @Override
    @Test
    public void tLoadHtmWalk()
    {
        super.tLoadHtmWalk();
    }

    @Override
    @Test
    public void tLoadPathWalk()
    {
        super.tLoadPathWalk();
    }

    @Override
    @Test
    public void tSavePath()
    {
        super.tSavePath();
    }

    @Override
    @Test
    public void tSaveWriter()
    {
        super.tSaveWriter();
    }

    @Test
    public void tEntityIsPreserved()
    {
    	// TODO: Get entities working
        //super.tEntityIsPreserved();
    }

    @Test
    public void tEntityIsPreservedSaved()
    {
    	// TODO: Get entities working
        //super.tEntityIsPreservedSaved();
    }

    @Test
    public void tEntityIsPreservedLoadSaved()
    {
    	// TODO: Get entities working
        //super.tEntityIsPreservedLoadSaved();
    }
}
